ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -719,14 +730,22 @@ private boolean verifyJoin(
               rightExpression
           ));
         }
+      } else {
+        // primary key join detected
+        verifyIfKeyTypesMatch(
```

Comment:
Side cleanup: pull this check out of the physical planner into the logical planner.

Need to do something similar later for other joins (ie, stream-stream and stream-table).